File: 2026/06-June/30-Tuesday/solution.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        # Track the most recent index of each character
        last_seen = {'a': -1, 'b': -1, 'c': -1}
        total_count = 0
        
        for i, char in enumerate(s):
            last_seen[char] = i
            # If all characters have been seen, min() will be >= 0
            total_count += min(last_seen.values()) + 1
            
        return total_count

File: 2026/06-June/30-Tuesday/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var numberOfSubstrings = function(s) {
    const lastSeen = { 'a': -1, 'b': -1, 'c': -1 };
    let totalCount = 0;
    
    for (let i = 0; i < s.length; i++) {
        lastSeen[s[i]] = i;
        
        // Accumulate valid starting positions
        totalCount += Math.min(lastSeen['a'], lastSeen['b'], lastSeen['c']) + 1;
    }
    
    return totalCount;
};
